x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/DataFrameDataExtractorFactory.java

```diff
@@ -57,35 +59,38 @@
     private final String analyticsId;
     private final String index;
     private final ExtractedFields extractedFields;
+    private final QueryBuilder query;
 
-    private DataFrameDataExtractorFactory(Client client, String analyticsId, String index, ExtractedFields extractedFields) {
+    private DataFrameDataExtractorFactory(Client client, String analyticsId, String index, ExtractedFields extractedFields,
+                                          QueryBuilder query) {
         this.client = Objects.requireNonNull(client);
         this.analyticsId = Objects.requireNonNull(analyticsId);
         this.index = Objects.requireNonNull(index);
         this.extractedFields = Objects.requireNonNull(extractedFields);
+        this.query = query;
     }
 
     public DataFrameDataExtractor newExtractor(boolean includeSource) {
         DataFrameDataExtractorContext context = new DataFrameDataExtractorContext(
                 analyticsId,
                 extractedFields,
                 Arrays.asList(index),
-                QueryBuilders.matchAllQuery(),
+                query == null ? QueryBuilders.matchAllQuery() : query,
```

Comment:
Opted for default value here. That way when the user calls `GET` on their config a matchall query is not magically added.

Reply:
Like I mentioned above, I would prefer setting the default on the config itself. I think it is better for the user to see all defaults and we've agreed to prefer that in a discussion of the past (though we don't always do it).